src/main/java/duke/tasks/Task.java

```diff
@@ -0,0 +1,76 @@
+package duke.tasks;
+
+public class Task {
+    protected enum Type {
+        T,
+        D,
+        E,
+    }
+
+    protected String description;
+    protected boolean isDone;
+    protected Type type;
+
+    /**
+     * Class Constructor
+     * @param description String Description of the Task
+     * @param type Type of the Task, ([T]oDo, [D]eadline, [E]vent)
+     */
+    public Task(String description, Type type) {
+        this.description = description;
+        this.isDone = false;
+        this.type = type;
+    }
+
+    /**
+     * Class Constructor taking in if the task is done
+     * @param description String Description of the Task
+     * @param type Type of the Task, ([T]oDo, [D]eadline, [E]vent)
+     * @param done boolean indicating if the task is done
+     */
+    public Task(String description, Type type, boolean done) {
+        this.type = type;
+        this.description = description;
+        this.isDone = done;
```

Comment:
Do we need to use this keyword here?

Reply:
Which keyword in particular are you referring to?